Assets/Scripts/LogicServices/Algorithm/SelfBuild.cs

```diff
@@ -1,19 +1,21 @@
+using System;
 using Models;
-using UnityEngine;
 
 namespace LogicServices.Algorithm
 {
     public class SelfBuild
     {
-        private readonly byte _selfBuildThreshold;
-        public SelfBuild(byte selfBuildThreshold)
+        public int SelfBuildThreshold { get; set; }
```

Comment:
private set?

Reply:
No, is public so it could be changed by the algorithm service when needed